### PR TITLE
Add sitemap

### PR DESCRIPTION
**Pull request by Carlgo11**
Found a useful script to generate sitemap.xml. Edited it to fit this project.

### CODE REVIEW

**Comment by Carlgo11** [comment]
Hmm file-extensions.sh doesn't like the sitemap.sh.options file :thinking: 
@phallobst how do you feel about adding `options` as a valid file extension to `/scripts`? I don't really have a better place to store the file :disappointed:

**Comment by phallobst** [comment]
I'm not sure why we'd nee the options file at all, since these are references only once in the script and could be used in-place.

If you really want to keep the file, why not just name it `sitemap.options.sh` instead? :wink:

**Comment by phallobst** [comment]
Is the `get_freqency` function any useful? The script will probably work on a freshly checked-out `git` workspace most of the time, i.e. the file timestamps will always lead to "daily" there, I reckon.

**Comment by Carlgo11** [comment]
I did a fetch and hard reset of my environment and it appears to work as expected there.
How do you propose we implement the options data into the shell script? I can't come up with a way that also is readable.

**Comment by phallobst** [comment]
I took the liberty to rename the options file, so that's one alternative. :wink:

This works as well:
```bash
OPTIONS='
-prune
! -iname robots.txt
! -iname manifest.json
! -iname service-worker.js
! -iname "sitemap.*"
! -iname ".*"
! -path "./css/*"
! -path "./js/*"
! -path "./img/*"
! -path "./entries/*"
'
cd "${1}" || exit 1
eval find . -type f $OPTIONS -printf '"%TY-%Tm-%Td%p\n"' | \
while read -r line; do
```

Or if you don't want to use eval:
```bash
cd "${1}" || exit 1
find . -type f \
-prune \
! -iname robots.txt \
! -iname manifest.json \
! -iname service-worker.js \
! -iname "sitemap.*" \
! -iname ".*" \
! -path "./css/*" \
! -path "./js/*" \
! -path "./img/*" \
! -path "./entries/*" \
-printf "%TY-%Tm-%Td%p\n" | \
while read -r line; do
```

**Comment by Carlgo11** [comment]
phallobst the Unix wizard :magic_wand: 
I added the eval suggestion :smile:

**Review by phallobst — APPROVED** [review]

File: scripts/sitemap.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# Copyright (c) 2017, Tino Reichardt <https://github.com/mcmilk/sitemap-generator>
+# Copyright (c) 2022, 2factorauth <https://github.com/2factorauth/twofactorauth>
+
+[[ $# -ne 1 ]] && { echo "USAGE: ./sitemap.sh _site" >&2; exit 1; }
+
+# url configuration
+URL="https://2fa.directory/"
+
+# values: always hourly daily weekly monthly yearly never
+get_freqency () {
```
Comment (dsoegijono) [inline scripts/sitemap.sh]:
```suggestion
get_frequency () {
```

File: scripts/sitemap.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+! -path "./img/*"
+! -path "./entries/*"
+'
+
+cd "${1}" || exit 1
+eval find . -type f $OPTIONS -printf '"%TY-%Tm-%Td%p\n"' | \
+while read -r line; do
+  DATE=${line:0:10}
+  FILE=${line:12}
+  get_freqency $((($(date +%s) - $(date +%s -r "${FILE}")) / 86400))
```
Comment (dsoegijono) [inline scripts/sitemap.sh]:
```suggestion
  get_frequency $((($(date +%s) - $(date +%s -r "${FILE}")) / 86400))
```

**Review by dsoegijono — APPROVED** [review]
Typo on "frequency" but otherwise lgtm

**Comment by Carlgo11** [comment]
@phallobst looks like -printf isn't an option in the busybox eval command.

**Comment by phallobst** [comment]
@Carlgo11
`-printf` is an option for _GNU find_, but the _busybox find_ does not support it. A slower workaround would be something like

```bash
eval find . -type f $OPTIONS -print0 | xargs -0 stat -c '%z%n' | \
while read -r line; do
  DATE=${line:0:10}
  FILE=${line:37}
```
or
```bash
eval find . -type f $OPTIONS -print0 | xargs -0 stat -c '%z%n' | \
while read -r line; do
  DATE=${line:0:10}
  FILE=${line:31}
```
depending on whether `stat` is the _busybox_ variant as well. The offsets for the `DATE` and `FILE` variables are correct on my system, but may need some adjustment.

**Pull request merged**